src/complex/Common/BoundingBox.hpp

```diff
@@ -81,145 +95,62 @@ class BoundingBox
    * @brief Returns the Min point
    * @return
    */
-  std::array<ValueType, 3> getMinPoint() const
+  PointType getMinPoint() const
```

Comment:
```suggestion
  const PointType& getMinPoint() const
```
Since we're changing the class anyways, I'd make the getters return by const ref.